### DISCUSSION TITLE
Do I need to treat boundary pores differently when simulating Fickian diffusion?

### DISCUSSION THREAD

**Discussion by ryanvolpi** [PMEAL/OpenPNM#1971] [Q&A]
I am trying to simulate diffusion through a wall. The image below is a sort of top down view, where the left side represents the inlet and the right side the outlet. I am running into some issues. I'm wondering if they are related to how the boundary pores on the left and right side are being handled. For example, [ this tutoral](https://notebook.community/PMEAL/OpenPNM/examples/tutorials/Intro%20to%20OpenPNM%20-%20Advanced) states:
> "The boundary pores we've added to the network should be treated a little bit differently. Specifically, they should have no volume or length (as they are not physically representative of real pores). To do this, we create two separate Geometry objects, one for internal pores and one for the boundaries" 

My question is: do I need to create a seperate geometry object for the boundary pores? And, if so, how do I make sure that these pores have no effect when simulating fickian diffusion? 

Top-down image:  ![image](https://user-images.githubusercontent.com/22488709/130111715-9d8a0621-6e1e-42c7-8688-42d93e6b82f5.png)

Thank you!

**Comment by ma-sadeghi** [comment]
Is this an extracted network? If so, did you extract it using PoreSpy?

**Reply by ryanvolpi** [reply]
Yes, using ps.networks.snow and specifying `boundary_faces=[inlet,outlet]`

**Reply by ma-sadeghi** [reply]
Then, I encourage you to upgrade your `porespy` to v2.0, then use the revamped version of SNOW algorithm (`porespy.networks.snow2`). It's pretty much the same, except it's much more robust and boundary pores work better with `openpnm`. I don't think you no longer need to add a separate geometry with `snow2`, the numbers should just work right out of the box (i.e., conductance of boundary throats should be automatically calculated to a large value, so they don't contribute to the overall resistance of the network). Let me know if it works for you.

**Reply by ryanvolpi** [reply]
Thank you for the support! Unfortunately, that brings me back to my original problem which is that I cannot figure out how to use `porespy.networks.snow2` in conjunction with `openpnm.algorithms.FickianDiffusion`. See: https://github.com/PMEAL/OpenPNM/discussions/1916. I've updated both packages to their most recent verions via pip and I still get the error **KeyError: 'throat.diffusive_conductance'**.

**Reply by ma-sadeghi** [reply]
Which `physics` object do you use? If `Standard`, try using `Basic`.

**Reply by ryanvolpi** [reply]
I was using Standard. Switching to Basic seemsto have solved the issue at hand as the model runs for some time whereas before it failed instantly. However, it still doesn't complete because after a few seconds the kernel (I'm using a jupyter notebook) dies.  I've tried the same procedure with a trivally small sample and get the same result.

**Reply by ma-sadeghi** [reply]
Hmm, could you share the code?

**Reply by ryanvolpi** [reply]
Here is a minimal reproduceable example
```
import porespy as ps
import openpnm as op

im = ps.generators.blobs(shape=[50, 50, 50], porosity=0.6)
snow_output = ps.networks.snow2(im, boundary_width=[3, 3, 3])
pn, geo = op.io.PoreSpy.import_data(snow_output.network)

air = op.phases.Air(network=pn)
phys_air = op.physics.Basic(network=pn, phase=air, geometry=geo)

h = pn.check_network_health()
op.topotools.trim(pn, pores=h['trim_pores'])
op.topotools.label_faces(pn)

fd = op.algorithms.FickianDiffusion(network=pn, phase=air)
fd.setup(phase=air)
fd.set_value_BC(pores=pn.pores("top"), values=1)
fd.set_value_BC(pores=pn.pores("bottom"), values=0)
fd.run()
```

**Reply by ma-sadeghi** [reply]
Thank you. I ran your code and indeed, it failed. It seems to be a bug in the way we do parallelization. To mitigate the issue for now, pass `parallelization=None` to `snow2` to skip that part (I checked and it successfully ran). Also, could you please open up an issue and report this bug? Thank you so much!

**Reply by ryanvolpi** [reply]
Are you saying I just I need to pass `parallelization=None` to `snow2`? I tried that and it did not resolve the issue. I have no problem with `snow2` running, just with `fd.run()` 

I created an issue to report the bug.

**Reply by ma-sadeghi** [reply]
Oh, I see. So, I don't think this is a bug. I wrote something for you in the issues page.
